docs/book/common_code/noderef_controller.md

```diff
@@ -0,0 +1,8 @@
+# Node Controller
+
+The noderef controller populates a `Machine`'s `Status.NodeRef` field using the Kubernetes Node
+object associated with the Machine. This controller supports only machines linked to a cluster.
+
+Infrastructure providers can opt-in to use this controller by providing a KubeConfig as a Kubernetes Secret.
+The secret must be stored in the namespace the Cluster lives in and named as `<cluster-name>-kubeconfig`,
+the data should only contain a single key called `value` and its data should be a valid Kubernetes KubeConfig.
```

Comment:
@chuckha looks like we're using `kubeconfig` in the upgrade tool but CAPI is expecting `value` - I guess we should switch the upgrade tool?